js/apps/account-ui/src/oid4vci/Oid4Vci.tsx

```diff
@@ -0,0 +1,218 @@
+    PageSection,
+    ActionList,
+    ActionListItem,
+    List,
+    ListItem,
+    MenuToggleElement,
+    MenuToggle
+} from '@patternfly/react-core';
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useAlerts } from "@keycloak/keycloak-ui-shared";
+import { usePromise } from "../utils/usePromise";
+import { Page } from "../components/page/Page";
+import { parseResponse } from "../api/parse-response";
+import { useEnvironment } from '../root/KeycloakContext';
+
+interface CredentialsIssuer {
+  credential_issuer: string;
+  credential_endpoint: string;
+  authorization_servers: string[]; 
+  credential_configurations_supported: Map<string, SupportedCredentialConfiguration>
+}
+
+interface SupportedCredentialConfiguration {
+  id: string,
+  format: string,
+  scope: string
+}
+
+type VCState = {
+  dropdownItems: string[],
+  selectOptions: Map<string, SupportedCredentialConfiguration>,
+  credentialIssuer?: CredentialsIssuer,
+  issuerDid: string,
+  qrCode: string,
+  isOpen: boolean,
+  offerQRVisible: boolean
+}
+
+export const Oid4Vci = () => {
+    const { addAlert, addError } = useAlerts();
+    const context = useEnvironment();
+   
+    const { t } = useTranslation();
+    const initialState: VCState = {
+      dropdownItems: [],
+      selectOptions: new Map<string, SupportedCredentialConfiguration>(),
+      issuerDid: "",
+      qrCode: "",
+      isOpen: false,
+      offerQRVisible: false
+    }
+    const initialSelected = t('verifiableCredentialsSelectionDefault') 
+
+    const [selected, setSelected] = useState<string>(initialSelected);
+    const [vcState, setState] = useState<VCState>(initialState);
+    const url:string = context.environment.authUrl
+    const realm = context.environment.realm
+    const wellKnownIssuer = url + "/realms/" + realm + "/.well-known/openid-credential-issuer"
+
+    usePromise(
+      (signal) =>
+        Promise.all([
+          getIssuer(wellKnownIssuer),
+        ]),
+      ([issuer]) => {
+        const ccsMap = new Map(Object.entries(issuer.credential_configurations_supported))
+        const ccsKeyArray = Array.from(ccsMap.keys());
+        setState({...vcState , credentialIssuer: issuer, dropdownItems: ccsKeyArray, selectOptions: ccsMap});
+    
+    });
+
+    useEffect(() => {
+      if(initialSelected !== selected){
+        requestVCOffer();
+      }
+    }, [selected]);
+
+    const getAccessToken =  async () => {
+      try {
+        await context.keycloak.updateToken();
+      } catch (error) {
+        await context.keycloak.login();
+      }
+      return context.keycloak.token;
+    }
+
+    const fetchWithToken = (url: string) => {
+      return getAccessToken()
+        .then(token => {
+          var options = {  
+            method: 'GET',
+            headers: {
+              'Authorization': 'Bearer ' +  token
+            }
+          }
+          return fetch(url, options)
+        })   
+    }
```

Comment:
We already have logic for this why don't you use that instead of creating your own

Reply:
Fixed in [0ba6aeb](https://github.com/keycloak/keycloak/pull/29174/commits/0ba6aebc92b9a1db47bd0443b2db6a015aaa762e)